test/spec/request-test.js

```diff
@@ -1425,4 +1427,263 @@ describe('Request-Promise', function () {
+                    name: 'resource2'
+                }
+            },
+            {
+                header: {
+                    'Last-Modified': now
+                    'Cache-Control': 'no-cache'
+                    'Content-Type': 'application/json'
+                }
+                content: {
+                    name: 'resource3'
+                }
+            },
+            {
+                header: {
+                    'Last-Modified': now
+                    'Cache-Control': 'no-cache'
+                    'Content-Type': 'application/json'
+                    'Age': 20
+                }
+                content: {
+                    name: 'resource4'
+                }
+            }
+        ]
+
+        var incrementResource = function(resource) {
+            resource.header['Last-Modified'].add(5, 'seconds')
+        }
+
+        before(function (done) {
+            // This creates a local cache server to test cache behavior of the lib
+            cacheServer = http.createServer(function (request, response) {
+                (bodyParser.json())(request, response, function () {
+                    var path = url.parse(request.url).pathname;
+                    var resourceName = parseInt(path.split('?')[0].split('/')[1]);
+                    var resourceIndex = resourceName[ resourceName.length - 1 ]
+                    var resource = resources[resourceIndex]
+                    if (resource){
+                        response.writeHead(200, resource.header);
+                        response.end(JSON.stringify(resource.content));
+                    } else {
+                        response.writeHead(404)
+                        reponse.end('')
+                    }
+                });
+            });
+
+            cacheServer.listen(5000, function () {
+                done();
+            });
+
+        });
+
+        after(function () {
+            cacheServer.close();
+        });
+
+        beforeEach(function () {
+            // require a fresh lib
+            delete require.cache[require.resolve('../../lib/rp.js')]
+            rp = require('../../lib/rp.js')
+
+            // reset the last modified of the resources
+            for (i=0; i<resources.length; i++){
+                resources[i].headers['Last-Modified'] = moment().format(LAST_MODIFIED_FORMAT)
+            }
+        });
+
+        assertResponse = function(response, statusCode, content, lastModified, age) {
+            expect(response.statuCode).to.be.equal(statusCode)
+            expect(JSON.parse(response.body).name).to.be.equal(content)
+            expect(response.headers['last-modified']).to.be.equal(now)
+            if (!age) {
+                expect(response.headers['age']).to.be.equal(age)
+            } else {
+                expect(response.headers['age'] > 0).to.be.true
+            }
+        }
+
+        it('should cache resource : cache-control, cache option true', function (done) {
```

Comment:
`cache` is a new option of the rp option object to control whether the cache feature is activated.
For the caching to happened `cache` must be set to true.
`cache` is false by default to avoid creating potential breaking change in the lib api.